pkg/controller/atlascluster/cluster.go

```diff
@@ -139,3 +141,31 @@ func ClustersEqual(log *zap.SugaredLogger, clusterAtlas mongodbatlas.Cluster, cl
 
 	return d == ""
 }
+
+func ensureConnectionSecrets(wctx *workflow.Context, k8sClient client.Client, project *mdbv1.AtlasProject, cluster *mongodbatlas.Cluster) workflow.Result {
+	databaseUsers := mdbv1.AtlasDatabaseUserList{}
+
+	for _, dbUser := range databaseUsers.Items {
+		found := false
+
+		for _, scope := range dbUser.GetScopes(mdbv1.ClusterScopeType) {
+			if scope == cluster.Name {
+				found = true
+				break
+			}
+
+			continue
+		}
+
+		if !found {
+			continue
+		}
+
+		result := atlasdatabaseuser.CreateOrUpdateConnectionSecrets(wctx, k8sClient, *project, dbUser)
```

Comment:
I think another condition should be that the user is "ready = true" what do you think? Otherwise if the user is not ready - this means that the connectivity to the cluster using this user may not work